src/client/feed/StoryContainer.js

```diff
@@ -0,0 +1,77 @@
+import _ from 'lodash';
+import { connect } from 'react-redux';
+import { push } from 'react-router-redux';
+import Story from '../components/Story/Story';
+import {
+  getAuthenticatedUser,
+  getPosts,
+  getBookmarks,
+  getPendingBookmarks,
+  getPendingLikes,
+  getRebloggedList,
+  getPendingReblogs,
+  getFollowingList,
+  getPendingFollows,
+  getIsEditorSaving,
+  getVotingPower,
+  getRewardFund,
+  getVotePercent,
+  getShowNSFWPosts,
+} from '../reducers';
+import { votePost } from '../post/postActions';
+import { toggleBookmark } from '../bookmarks/bookmarksActions';
+import { editPost } from '../post/Write/editorActions';
+import { reblog } from '../app/Reblog/reblogActions';
+import { followUser, unfollowUser } from '../user/userActions';
+
+const mapStateToProps = (state, { id }) => {
+  const user = getAuthenticatedUser(state);
+  const post = getPosts(state)[id];
+
+  const userVote = _.find(post.active_votes, { voter: user.name }) || {};
+
+  const postState = {
+    isReblogged: getRebloggedList(state).includes(post.id),
+    isReblogging: getPendingReblogs(state).includes(post.id),
+    isSaved: !!getBookmarks(state)[post.id],
+    isLiked: userVote.percent > 0,
+    isReported: userVote.percent < 0,
+    userFollowed: getFollowingList(state).includes(post.author),
+  };
+
+  const pendingLikes = getPendingLikes(state);
+
+  const pendingLike =
+    pendingLikes[post.id] &&
```

Comment:
`pendingLikes[post.id]` is repeated multiple times from here until line 50, maybe it would be a good idea to instantiate it like `const pendingLike = _.get(pendingLikes, post.id, {})`